Add migrations for core and assessment databases

maximilianoertel opened the pull request:
## Proposed changes


## Types of changes
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (non-breaking change that does not add functionality but makes code cleaner or more efficient)
- [ ] Documentation Update
- [ ] Tests (new or updated tests)
- [ ] Style (changes to code styling)
- [ ] CI (continuous integration changes)
- [ ] Repository Maintenance
- [ ] Other (please describe below)

- Resolves https://github.com/yeatmanlab/roar-project-management/issues/1186

Emily-ejag reviewed (approved):
It took a long time to review, but it looks good to me :)

richford commented on apps/backend/migrations/core/0036_add_rostering_provider_ids_table.sql:
```diff
@@ -0,0 +1,12 @@
+CREATE TABLE "app"."rostering_provider_ids" (
+	"provider_type" "app"."rostering_provider" NOT NULL,
```
```suggestion
    "partner_id" text NOT NULL,
	"provider_type" "app"."rostering_provider" NOT NULL,
```

maximilianoertel commented:
@richford I've added the missing `partnerId` to the `rostering_provider_ids`. Given yesterday's discussion where you explained that some rostering providers have globally unique IDs, while others only have unique IDs within a partner context,  I've also updated the constraints and primary key to match the expected behaviour
